pat/consumer_tests/step_6.py

```diff
@@ -0,0 +1,699 @@
+    """The Reference Provider invokes Activate:
+    - Payload: 3 arguments
+    - The Reference Provider immediately responds with Fin
+    - Action: The Reference Provider accepts 3 arguments, concatenates them and writes them to the operation target's metric value
+    """  # noqa: D205, D400, D415, E501, W505
+    step = f'{__STEP__}f'
+    test_results: list[bool] = []
+
+    # TODO: iterate over operations  # noqa: FIX002, TD002, TD003
+    string_argument = msg_types.Argument()
+    string_argument.ArgValue = ''.join(random.choice(string.ascii_lowercase) for i in range(20))
+    decimal_argument = msg_types.Argument()
+    decimal_argument.ArgValue = str(random.randint(1, 2**20))
+    any_uri_argument = msg_types.Argument()
+    any_uri_argument.ArgValue = uuid.uuid4().urn
+    arguments = [string_argument, decimal_argument, any_uri_argument]
+    activate_operation_handle = 'activate_1.sco.mds_0'
+    try:
+        activate_operation: descriptorcontainers.ActivateOperationDescriptorContainer = (
+            consumer.mdib.descriptions.handle.get_one(activate_operation_handle)
+        )
+    except KeyError:
+        logger.error(  # noqa: TRY400
+            'no operation with handle "%s" found in MDIB',
+            activate_operation_handle,
+            extra={'step': step},
+        )
+        return False
+    expected_value = ''.join([argument.ArgValue for argument in arguments])
+    observed = threading.Event()
+    if activate_operation.MaxTimeToFinish is None:
+        timeout = 10.0
+        logger.warning(
+            'Activate operation with the handle %s has no MaxTimeToFinish, using default of %d seconds',
+            activate_operation.Handle,
+            timeout,
+            extra={'step': step},
+        )
+    else:
+        timeout = activate_operation.MaxTimeToFinish
+
+    def _observe_operation_metric_value(metrics_by_handle: dict):
+        if activate_operation.OperationTarget in metrics_by_handle:
+            state = metrics_by_handle[activate_operation.OperationTarget]
+            logger.debug(
+                'received an update for "%s" with value "%s"',
+                activate_operation.OperationTarget,
+                state.MetricValue,
+                extra={'step': step},
+            )
+            if state.MetricValue is not None and state.MetricValue.Value == expected_value:
+                observed.set()
+
+    with observables.bound_context(consumer.mdib, metrics_by_handle=_observe_operation_metric_value):
+        fut: Future = consumer.set_service_client.activate(activate_operation_handle, arguments)
+        try:
+            operation_result = fut.result(timeout)
+        except TimeoutError:
+            logger.exception(
+                'Activate operation not finished within the timeout of %s seconds',
+                activate_operation.MaxTimeToFinish,
+                extra={'step': step},
+            )
+            return False
+        if observed.wait(timeout):
```

Comment:
in 6e another timeout was used for the observer. would suggest to make it consistent

Reply:
This is not yet solved. Make the calculation for the timeout as done in 6e

```python
    timeout = timeout + 1 - (time.perf_counter() - start)
    observed.wait(timeout)
```